### PR TITLE
add starknet_estimateMessageFee endpoint

### PR DESCRIPTION
**Pull request by ArielElp**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/100)

### CODE REVIEW

File: api/starknet_api_openrpc.json [hunk 1]
```diff
@@ -536,6 +536,56 @@
...
+        {
+            "name": "starknet_estimateMessageFee",
+            "summary": "estimate the L2 fee of a message sent on L1",
+            "description": "estimates the resources required by the l1_handler transaction induced by the message",
+            "params": [
+                {
+                    "name": "message",
+                    "description": "the message's parameters",
+                    "schema": {
+                        "$ref": "#/components/schemas/FUNCTION_CALL"
```
Comment (kkovaacs) [inline api/starknet_api_openrpc.json:548]: Wouldn't it be more precise if we did not reuse FUNCTION_CALL here? Just similar to how there are separate types for the L1 handler call and the function call in cairo-lang: [CallFunction](https://github.com/starkware-libs/cairo-lang/blob/361fe32d5930db340ea78fe05aedfe706f6c9405/src/starkware/starknet/services/api/feeder_gateway/request_objects.py#L13) and [CallL1Handler](https://github.com/starkware-libs/cairo-lang/blob/361fe32d5930db340ea78fe05aedfe706f6c9405/src/starkware/starknet/services/api/feeder_gateway/request_objects.py#L32)

For example reusing `calldata` in FUNCTION_CALL for storing the `payload` of the L1 message sounds a bit like a hack to me...

**Review by ArielElp — COMMENTED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/100)* status: all files reviewed (commit messages unreviewed), 1 unresolved discussion (waiting on @kkovaacs)

___
*[`api/starknet_api_openrpc.json` line 548 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/100#gh-1221302073:-NYRqkOGBtbK_fEOpKQL:b-aviggp) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/37c5ed1b5e4608cf8b80b3e06c726aadfa4f0db6/api/starknet_api_openrpc.json#L548)):*
<details><summary><i>Previously, kkovaacs (KOVACS Krisztian) wrote…</i></summary><blockquote>

Wouldn't it be more precise if we did not reuse FUNCTION_CALL here? Just similar to how there are separate types for the L1 handler call and the function call in cairo-lang: [CallFunction](https://github.com/starkware-libs/cairo-lang/blob/361fe32d5930db340ea78fe05aedfe706f6c9405/src/starkware/starknet/services/api/feeder_gateway/request_objects.py#L13) and [CallL1Handler](https://github.com/starkware-libs/cairo-lang/blob/361fe32d5930db340ea78fe05aedfe706f6c9405/src/starkware/starknet/services/api/feeder_gateway/request_objects.py#L32)

For example reusing `calldata` in FUNCTION_CALL for storing the `payload` of the L1 message sounds a bit like a hack to me...
</blockquote></details>

added a MSG_FROM_L1 type to distinguish this from function call and not use calldata

File: api/starknet_api_openrpc.json [hunk 2]
```diff
@@ -536,6 +536,56 @@
...
+                {
+                    "name": "message",
+                    "description": "the message's parameters",
+                    "schema": {
+                        "$ref": "#/components/schemas/MSG_FROM_L1"
+                    },
+                    "required": true
+                },
+                {
+                    "name": "sender_address",
```
Comment (kkovaacs) [inline api/starknet_api_openrpc.json:553]: Isn't `sender_address` redundant now that `from_address` is part of `MSG_FROM_L1`?

**Review by ArielElp — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/100)* status: all files reviewed, 2 unresolved discussions (waiting on @kkovaacs)

___
*[`api/starknet_api_openrpc.json` line 553 at r2](https://reviewable.io/reviews/starkware-libs/starknet-specs/100#gh-1252952796:-NZdfgF50O_EakINPWjt:b98aj3v) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/bd5ea0055c8eab7caf6a1fa5e75d1da7063291ef/api/starknet_api_openrpc.json#L553)):*
<details><summary><i>Previously, kkovaacs (KOVACS Krisztian) wrote…</i></summary><blockquote>

Isn't `sender_address` redundant now that `from_address` is part of `MSG_FROM_L1`?
</blockquote></details>

Correct, fixed

**Review by kkovaacs — APPROVED** [review]

**Review by kkovaacs — COMMENTED** [review]
LGTM

> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/100#-:-NZegvlvAq9uOF1-4_zh:bb6ywmx)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @ArielElp)

**Pull request merged**